src/pyfaf/queries.py

```diff
@@ -1389,30 +1344,150 @@ def get_all_report_hashes(db, date_from=None,
+    db.session.query(st.BzBugCc).filter(st.BzBugCc.bug_id == bug_id).delete(False)
+    db.session.query(st.BzBugHistory).filter(st.BzBugHistory.bug_id == bug_id).delete(False)
+    db.session.query(st.BzAttachment).filter(st.BzAttachment.bug_id == bug_id).delete(False)
+    db.session.query(st.ReportBz).filter(st.ReportBz.bzbug_id == bug_id).delete(False)
+    db.session.query(st.BzBug).filter(st.BzBug.id == bug_id).delete(False)
+
+def get_reportcontactmails_by_id(db, contact_email_id):
+    """
+    Return a query for ReportContactEmail objects for given contact_email_id
+    """
+    return (db.session.query(st.ReportContactEmail)
+            .filter(st.ReportContactEmail.contact_email_id == contact_email_id))
+
+def delete_report(db, report_id):
+    """
+    Delete report and all records in tables associated with the report for given report_id.
+    """
+    query = db.session.query
+
+    query(st.ReportArchive).filter(st.ReportArchive.report_id == report_id).delete(False)
+    query(st.ReportArch).filter(st.ReportArch.report_id == report_id).delete(False)
+    query(st.ReportComment).filter(st.ReportComment.report_id == report_id).delete(False)
+    query(st.ReportContactEmail).filter(st.ReportContactEmail.report_id == report_id).delete(False)
+    query(st.ReportExecutable).filter(st.ReportExecutable.report_id == report_id).delete(False)
+    query(st.ReportExternalFaf).filter(st.ReportExternalFaf.report_id == report_id).delete(False)
+    query(st.ReportHash).filter(st.ReportHash.report_id == report_id).delete(False)
+    query(st.ReportHistoryDaily).filter(st.ReportHistoryDaily.report_id == report_id).delete(False)
+    query(st.ReportHistoryMonthly).filter(st.ReportHistoryMonthly.report_id == report_id).delete(False)
+    query(st.ReportHistoryWeekly).filter(st.ReportHistoryWeekly.report_id == report_id).delete(False)
+    query(st.ReportOpSysRelease).filter(st.ReportOpSysRelease.report_id == report_id).delete(False)
+    query(st.ReportPackage).filter(st.ReportPackage.report_id == report_id).delete(False)
+    query(st.ReportRaw).filter(st.ReportRaw.report_id == report_id).delete(False)
+    query(st.ReportReason).filter(st.ReportReason.report_id == report_id).delete(False)
+    query(st.ReportReleaseDesktop).filter(st.ReportReleaseDesktop.report_id == report_id).delete(False)
+    query(st.ReportSelinuxContext).filter(st.ReportSelinuxContext.report_id == report_id).delete(False)
+    query(st.ReportSelinuxMode).filter(st.ReportSelinuxMode.report_id == report_id).delete(False)
+    query(st.ReportUnknownPackage).filter(st.ReportUnknownPackage.report_id == report_id).delete(False)
+    query(st.ReportUptime).filter(st.ReportUptime.report_id == report_id).delete(False)
+    query(st.ReportURL).filter(st.ReportURL.report_id == report_id).delete(False)
+
+    # delete backtraces
+    bt = query(st.ReportBacktrace).filter(st.ReportBacktrace.report_id == report_id).first()
```

Comment:
Are we sure, that there will be always one bt? If there will be none, the following line will fail (None.id) and if there are more than one, only one bt will be deleted. The same with thread.